setup.py

```diff
@@ -10,6 +15,44 @@
     exec(file.read())
 
 
+def get_long_description():
```

Comment:
It took me a while to grok what this is doing. Nicely done. Looks very thorough. Where have you been able to test it (on which operating systems, I mean)?